READMEs/c2d-flow.md

```diff
@@ -0,0 +1,381 @@
+
+#needed to mint fake OCEAN for testing with ganache
+export FACTORY_DEPLOYER_PRIVATE_KEY=0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58
+
+```
+
+### Config in Python
+
+In the work console:
+```console
+python
+```
+
+For the following steps, we use the Python console. Keep it open between steps.
+
+In the Python console:
+```python
+#create ocean instance
+from ocean_lib.config import Config
+from ocean_lib.ocean.ocean import Ocean
+config = Config('config.ini')
+ocean = Ocean(config)
+
+print(f"config.network_url = '{config.network_url}'")
+print(f"config.metadata_cache_uri = '{config.metadata_cache_uri}'")
+print(f"config.provider_url = '{config.provider_url}'")
+
+# Create Alice's wallet
+import os
+from ocean_lib.web3_internal.wallet import Wallet
+alice_wallet = Wallet(ocean.web3, private_key=os.getenv('TEST_PRIVATE_KEY1'))
+print(f"alice_wallet.address = '{alice_wallet.address}'")
+
+# Mint OCEAN for ganache only
+from ocean_lib.ocean.mint_fake_ocean import mint_fake_OCEAN
+mint_fake_OCEAN(config)
+
+assert alice_wallet.web3.eth.get_balance(alice_wallet.address) > 0, "need ETH"
+```
+
+## 2. Alice publishes data asset
+
+In the same Python console:
+```python
+# Publish DATA datatoken, mint tokens
+from ocean_lib.web3_internal.currency import to_wei
+
+DATA_datatoken = ocean.create_data_token('DATA1', 'DATA1', alice_wallet, blob=ocean.config.metadata_cache_uri)
+DATA_datatoken.mint(alice_wallet.address, to_wei(100), alice_wallet)
+print(f"DATA_datatoken.address = '{DATA_datatoken.address}'")
+
+# Specify metadata & service attributes for Branin test dataset.
+# It's specified using _local_ DDO metadata format; Aquarius will convert it to remote
+# by removing `url` and adding `encryptedFiles` field.
+DATA_metadata = {
+    "main": {
+        "type": "dataset",
+        "files": [
+	  {
+	    "url": "https://raw.githubusercontent.com/trentmc/branin/main/branin.arff",
+	    "index": 0,
+	    "contentType": "text/text"
+	  }
+	],
+	"name": "branin", "author": "Trent", "license": "CC0",
+	"dateCreated": "2019-12-28T10:55:11Z"
+    }
+}
+DATA_service_attributes = {
+    "main": {
+        "name": "DATA_dataAssetAccessServiceAgreement",
+        "creator": alice_wallet.address,
+        "timeout": 3600 * 24,
+        "datePublished": "2019-12-28T10:55:11Z",
+        "cost": 1.0, # <don't change, this is obsolete>
+        }
+    }
+
+# Set up a service provider. We'll use this same provider for ALG
+from ocean_lib.data_provider.data_service_provider import DataServiceProvider
+provider_url = DataServiceProvider.get_url(ocean.config)
+# returns "http://localhost:8030"
+
+# Calc DATA service compute descriptor
+from ocean_lib.common.agreements.service_factory import ServiceDescriptor
+DATA_compute_service_descriptor = ServiceDescriptor.compute_service_descriptor(DATA_service_attributes, provider_url)
+
+#Publish metadata and service info on-chain
+DATA_ddo = ocean.assets.create(
```

Comment:
FYI: I got error here when running with Barge with `Aquarius` v3.1.1, But when I downgraded `Aquarius` to v3.0.1, the same instruction worked.
Error said the Datatoken was already associated with an asset although that was not true. Unfortunately, I dont have exact logs now. I can reproduce them if needed,

Reply:
That was ocean.py not being in sync with the Aquarius API. Fixed now with the new release.